traffic_ops/traffic_ops_golang/deliveryservice/autorenewcerts.go

```diff
@@ -102,14 +102,14 @@ func renewCertificates(w http.ResponseWriter, r *http.Request, deprecated bool)
 		existingCerts = append(existingCerts, ExistingCerts{Version: ds.Version, XmlId: ds.XmlId})
 	}
 
-	ctx, _ := context.WithTimeout(r.Context(), AcmeTimeout*time.Duration(len(existingCerts)))
+	ctx, cancelTx := context.WithTimeout(r.Context(), AcmeTimeout*time.Duration(len(existingCerts)))
 
 	asyncStatusId, errCode, userErr, sysErr := api.InsertAsyncStatus(inf.Tx.Tx, "ACME async job has started.")
 	if userErr != nil || sysErr != nil {
 		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
```

Comment:
I know you didn't write the code, but I think this is meant to `return` here, in which case we should also `cancel()`. I think L100 is also missing a `return`

Reply:
Good catch, cancelling and returning in d8ae1cc61c.